### PR TITLE
KOGITO-7119: Establish test runner for serverless tooling in vscode

### PR DESCRIPTION
**Pull request by domhanak**
Adds first testing project greeting-flow
Adds first smoke test with example
Add helpers to work with serverless files in VSCode
Add helpers for asserts and SWF editor and text editor.

@tomasdavidorg  ready for inital checks,  I will add some polish later.

### CODE REVIEW

File: packages/vscode-extension-serverless-workflow-editor/it-tests/helpers/CommonAsserts.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates.
```
Comment (tomasdavidorg) [inline packages/vscode-extension-serverless-workflow-editor/it-tests/helpers/CommonAsserts.ts]: Also in other files:
```suggestion
 * Copyright 2022 Red Hat, Inc. and/or its affiliates.
```

File: packages/vscode-extension-serverless-workflow-editor/it-tests/resources/greeting-flow/README.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
+# greeting-flow Project
```
Comment (tomasdavidorg) [inline packages/vscode-extension-serverless-workflow-editor/it-tests/resources/greeting-flow/README.md]: Do we need all the greeting-flow project files?
We don't run the project directly so the README is probably not necessary. 
Are the docker files needed? I would say they are not necessary for the tests.

File: packages/vscode-extension-serverless-workflow-editor/it-tests/resources/greeting-flow/src/main/resources/META-INF/resources/index.html [hunk 3]
```diff
@@ -0,0 +1,351 @@
+<!DOCTYPE html>
```
Comment (tomasdavidorg) [inline packages/vscode-extension-serverless-workflow-editor/it-tests/resources/greeting-flow/src/main/resources/META-INF/resources/index.html]: This file is also probably not needed.

File: packages/vscode-extension-serverless-workflow-editor/it-tests/settings.json [hunk 4]
```diff
@@ -0,0 +1,12 @@
+{
```
Comment (tomasdavidorg) [inline packages/vscode-extension-serverless-workflow-editor/it-tests/settings.json:1]: Are these settings used for the tests?

File: packages/vscode-extension-serverless-workflow-editor/it-tests/serverless-workflow-editor-extension-smoke.test.ts [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    const swfEditor = new SwfEditorTestHelper(editorWebviews[1]);
+    const swfTextEditor = new SwfTextEditorTestHelper(editorWebviews[0]);
+
+    const textEditor = await swfTextEditor.getSwfTextEditor();
+    // check that something is displayed
+    assert.equal(await textEditor.getNumberOfLines(), 81);
+
+    // fined elements, asserts they exist
+    const svg = await swfEditor.getSvgElement();
+    const mermaid = await swfEditor.getMermeaidDivElement();
```
Comment (tomasdavidorg) [inline packages/vscode-extension-serverless-workflow-editor/it-tests/serverless-workflow-editor-extension-smoke.test.ts]: These variables are not used anywhere.

File: packages/vscode-extension-serverless-workflow-editor/it-tests/resources/greeting-flow/src/main/resources/greetings.sw.json.svg [hunk 6]
```diff
@@ -0,0 +1 @@
...n class="nodeLabel">type = Inject State</span></div></foreignObject></g></g><g class="node statediagram-state" id="state-GetGreeting-8" transform="translate(432.50390625, 432.75)"><rect class="outer title-state" x="-97.5625" y="-61.75" width="195.125" height="123.5"></rect><line class="divider" x1="-97.5625" x2="97.5625" y1="-30.25" y2="-30.25"></line><g class="label" transform="translate(-90.0625, -58.75)"><foreignObject width="87.15625" height="24" transform="translate( 46.484375, 0)"><div xmlns="http://www.w3.org/1999/xhtml" style="display: inline-block; white-space: nowrap;"><span class="nodeLabel">GetGreeting</span></div></foreignObject><foreignObject width="180.125" height="72" transform="translate( 0, 36.5)"><div xmlns="http://www.w3.org/1999/xhtml" style="display: inline-block; white-space: nowrap;"><span class="nodeLabel">type = Operation State<br/>Action mode = sequential<br/>Num. of actions = 1</span></div></foreignObject></g></g><g class="node default" id="state-root_end-8" transform="translate(432.50390625, 551.5)"><circle class="state-start" r="7" width="14" height="14"></circle><circle class="state-end" r="5" width="10" height="10"></circle></g></g></g></g></g></svg>
```
Comment (tomasdavidorg) [inline packages/vscode-extension-serverless-workflow-editor/it-tests/resources/greeting-flow/src/main/resources/greetings.sw.json.svg]: Is this just exported Svg? Is it used in the test?

File: packages/vscode-extension-serverless-workflow-editor/it-tests/serverless-workflow-editor-extension-smoke.test.ts [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+    await testHelper.closeAllNotifications();
+  });
+
+  afterEach(async function () {
+    this.timeout(15000);
+    try {
+      await testHelper.closeAllEditors();
+      await testHelper.closeAllNotifications();
+    } catch (e) {
+      // noop
```
Comment (tomasdavidorg) [inline packages/vscode-extension-serverless-workflow-editor/it-tests/serverless-workflow-editor-extension-smoke.test.ts]: We should probably at least log/print the exceptions.

File: packages/vscode-extension-serverless-workflow-editor/it-tests/resources/greeting-flow/README.md (hunk 2 above)
Comment (domhanak) [reply packages/vscode-extension-serverless-workflow-editor/it-tests/resources/greeting-flow/README.md]: My aim was to introduce a rule that this project must be buildable after smoke tests. Hence keeping al files, but will test it and remove what is nto needed for build test.

File: packages/vscode-extension-serverless-workflow-editor/it-tests/settings.json (hunk 4 above)
Comment (domhanak) [reply packages/vscode-extension-serverless-workflow-editor/it-tests/settings.json:1]: Yes, see package.json script

File: packages/vscode-extension-serverless-workflow-editor/it-tests/resources/greeting-flow/src/main/resources/greetings.sw.json.svg (hunk 6 above)
Comment (domhanak) [reply packages/vscode-extension-serverless-workflow-editor/it-tests/resources/greeting-flow/src/main/resources/greetings.sw.json.svg]: Not yet, plan is to introduce a rule that svg must exist for each opened file as it is how extension works.

File: packages/vscode-extension-serverless-workflow-editor/it-tests/serverless-workflow-editor-extension-smoke.test.ts [hunk 8]
```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates.
```
Comment (tomasdavidorg) [inline packages/vscode-extension-serverless-workflow-editor/it-tests/serverless-workflow-editor-extension-smoke.test.ts]:
```suggestion
 * Copyright 2022 Red Hat, Inc. and/or its affiliates.
```

**Review by tomasdavidorg — APPROVED** [review]

**Comment by domhanak** [comment]
@tomasdavidorg @tiagobento modified the smoke test to be more lenient and useful, and added it to build commands, let me know if still ok.

**Review by tomasdavidorg — APPROVED** [review]

**Comment by domhanak** [comment]
Disabled the tests until #1052 is merged. @tiagobento Please merge if ok. There is no impact on builds and allows other team members from qe to work on more tests in the meantime.

**Review by yesamer — APPROVED** [review]
@domhanak Approved. As we discussed in chat, a future extension of this work is to define a separate module which contains commons utils tests assets, to be reused in projects like `swf`, `yard`, and kogito editors. Thank you.

**Review by yesamer — APPROVED** [review]

**Pull request merged**